### ISSUE TITLE
Add "possessive" functionality to the Str helper

### ISSUE THREAD

**Issue by tkaravou** [laravel/ideas#2168]
In several projects that I've worked on, I've had to implement a possessive helper. I think it would be valuable to have this as part the core Str helper.

Examples of usage
`Str::possessive('Billy')` outputs `Billy's`
`Str::possessive('Steves')` outputs `Steves'`
`Str::possessive('marko2020')` outputs `marko2020's`

**Comment by JamesPoel** [comment]
+1, however the ternary should be in brackets, right?

Otherwise this will only return empty string or "s".

`return "$subject'" . ( self::endsWith($subject, ['s', 'S']) ? '' : 's' );`

**Comment by tkaravou** [comment]
That's right James, I actually fixed it in my project. Thanks for pointing it out, I'll update the main post.
